[CI] FullClusterRestartIT testWatcherWithApiKey {cluster=UPGRADED} failing

**Build Scans:**
- [elasticsearch-intake #17575 / 8.16.4_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mmd4eikn5ahty)
- [elasticsearch-pull-request #55460 / 9.1.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/x4di3hyakxqey)

**Reproduction Line:**
```
./gradlew ":x-pack:qa:full-cluster-restart:v8.16.4#bwcTest" -Dtests.class="org.elasticsearch.xpack.restart.FullClusterRestartIT" -Dtests.method="testWatcherWithApiKey {cluster=UPGRADED}" -Dtests.seed=F9421E65C44C4430 -Dtests.bwc=true -Dtests.locale=ebu -Dtests.timezone=Etc/GMT+11 -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.restart.FullClusterRestartIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testWatcherWithApiKey%20%7Bcluster%3DUPGRADED%7D),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: version increased: [true], executed: [false]
Expected: is <true>
     but: was <false>
```

**Issue Reasons:**
- [main] 2 failures in test testWatcherWithApiKey {cluster=UPGRADED} (0.2% fail rate in 903 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -248,9 +248,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/121625
 - class: org.elasticsearch.xpack.application.CohereServiceUpgradeIT
   issue: https://github.com/elastic/elasticsearch/issues/121537
-- class: org.elasticsearch.xpack.restart.FullClusterRestartIT
-  method: testWatcherWithApiKey {cluster=UPGRADED}
-  issue: https://github.com/elastic/elasticsearch/issues/122061
 - class: org.elasticsearch.test.rest.ClientYamlTestSuiteIT
   method: test {yaml=snapshot.delete/10_basic/Delete a snapshot asynchronously}
   issue: https://github.com/elastic/elasticsearch/issues/122102

